<!DOCTYPE html>
<html>

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="shortcut icon" href="../img/favicon.ico">
    <title>Example: creating and using an LCA database for taxonomic analysis &mdash; sourmash examples</title>
    <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato:400,700">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/gh/tonsky/FiraCode@1.206/distr/fira_code.css">
    <link rel="stylesheet" href="//use.fontawesome.com/releases/v5.8.1/css/all.css">
    <link rel="stylesheet" href="//use.fontawesome.com/releases/v5.8.1/css/v4-shims.css">
    <link rel="stylesheet" href="../css/theme.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
    <script src="//code.jquery.com/jquery-2.1.1.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
    <script>
        hljs.initHighlightingOnLoad();
    </script> 
</head>

<body ontouchstart="">
    <div id="container">
        <aside>
            <div class="home">
                <div class="title">
                    <button class="hamburger"></button>
                    <a href=".." class="site-name"> sourmash examples</a>
                </div>
                <div class="search">
                    <div role="search">
    <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
        <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
    </form>
</div>
                </div>
            </div>
            <nav class="nav">
                <ul class="root">
                    <li class="toctree-l1"><a class="nav-item" href="..">Home</a></li>
                    <li class="toctree-l1"><button class="section nav-item">Examples</button>
<ul class="subnav">
    <li class="toctree-l2"><a class="nav-item" href="../9-install-sourmash/">Example: install sourmash</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../6-creating-a-conda-environment-with-the-latest-version-of-sourmash/">Example: creating a conda environment with the latest version of sourmash</a></li>
    <li class="toctree-l2 current"><a class="nav-item current" href="./">Example: creating and using an LCA database for taxonomic analysis</a>
<ul class="subnav">
</ul></li>
    <li class="toctree-l2"><a class="nav-item" href="../1-downloading-sketching-and-searching-a-collection-of-FASTA-files/">Example: downloading sketching and searching a collection of FASTA files</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../5-searching-collection-manifests-directly/">Example: searching collection manifests directly</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../10-test-new-example/">Example: test new example</a><a class="nav-item" href="../10-test-new-example/">Example: test new example</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../7-using-genomeupdater.sh-and-sketch-fromfile-to-build-a-custom-database/">Example: using genome_updater.sh and sketch fromfile to build a custom database</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../3-using-sourmash-tax-to-classify-a-metagenome/">Example: using sourmash tax to classify a metagenome</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../4-using-picklists-and-manifests-to-work-with-a-small-subset-of-a-large-database/">Example: using picklists and manifests to work with a small subset of a large database</a></li>
</ul></li>
                    <li class="toctree-l1"><button class="section nav-item">Categories</button>
<ul class="subnav">
    <li class="toctree-l2"><a class="nav-item" href="../l-advanced/">advanced examples</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-databases/">creating databases</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-developer/">examples for developers</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-manifests/">examples of using manifests</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-taxonomy/">examples of using taxonomy</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-needs-improvement/">examples that need improvement</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-gtdb-rs207/">examples using GTDB RS207</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-picklists/">examples using picklists</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-sourmash-v4.2/">examples using sourmash v4.2 functionality</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-sourmash-v4.4/">examples using sourmash v4.4 functionality</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-sourmash-v4.0/">examples using sourmash v4.x</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-intermediate/">intermediate examples</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-intro/">introductory examples</a></li>
    <li class="toctree-l2"><a class="nav-item" href="../l-fasta/">working with FASTA files</a></li>
</ul></li>
                </ul>
            </nav>
            <div class="repo">
    <div class="previous"><a href="../6-creating-a-conda-environment-with-the-latest-version-of-sourmash/">&laquo; Previous</a></div>
    <div class="next"><a href="../1-downloading-sketching-and-searching-a-collection-of-FASTA-files/">Next &raquo;</a></div>
</div>
        </aside>
        <div id="spacer"><button class="arrow"></button></div>
        <main>
            <div class="home-top">
                <button class="hamburger"></button>
                <a href=".." class="site-name"> sourmash examples</a>
            </div>
            <div id="main">
                <nav class="breadcrumbs">
<ul>
    <li>Examples</li>
</ul>
</nav>
                <div id="content"><h1 id="example-creating-and-using-an-lca-database-for-taxonomic-analysis">Example: creating and using an LCA database for taxonomic analysis</h1>
<p><em><a href="https://github.com/sourmash-bio/sourmash-examples/issues/2">sourmash-bio/sourmash-examples#2</a></em></p>
<hr />
<p>First, download and sketch 64 genomes from <a href="https://www.biorxiv.org/content/10.1101/155358v3">Awad et al., 2017</a> using the instructions in <a href="../1-downloading-sketching-and-searching-a-collection-of-FASTA-files/">Example: downloading, sketching, and searching a collection of FASTA files</a>. You'll need <code>podar-ref.zip</code>.</p>
<p>Next, you'll need <a href="https://osf.io/4yhjw/"><code>podar-lineage.csv</code></a> -</p>
<pre><code class="language-shell">curl -L https://osf.io/4yhjw/download -o podar-lineage.csv
</code></pre>
<p>Now create an LCA database in SQL format:</p>
<pre><code class="language-shell">sourmash lca index podar-lineage.csv podar-ref.lca.sql podar-ref.zip \
    -F sql -C 3 --split-identifiers
</code></pre>
<p>Extract one of the Shewanella genomes from <code>podar-ref.zip</code> using <code>sourmash sig grep</code>:</p>
<pre><code class="language-shell">sourmash sig grep OS223 podar-ref.zip -o shew-os223.sig
</code></pre>
<p>and now you can classify genomes with <code>lca classify</code>:</p>
<pre><code class="language-shell">sourmash lca classify --query shew-os223.sig --db podar-ref.lca.sql
</code></pre>
<p>and you should see it classified correctly:</p>
<pre><code>&quot;NC_011663.1 Shewanella baltica OS223, complete genome&quot;,found,Bacteria,Proteobacteria,Gammaproteobacteria,Alteromonadales,Shewanellaceae,Shewanella,Shewanella baltica,Shewanella baltica OS223
</code></pre>
<p>You can use <code>lca summarize</code> to classify the genome as if it were a metagenome mixture, too:</p>
<pre><code>sourmash lca summarize --query shew-os223.sig --db podar-ref.lca.sql
</code></pre>
<p>and you should see:</p>
<pre><code>50.5%   278   Bacteria;Proteobacteria;Gammaproteobacteria;Alteromonadales;Shewanellaceae;Shewanella;Shewanella baltica;Shewanella baltica OS223   shew-os223.sig:38729c63 NC_011663.1 Shewanella baltica OS223, complete genome
100.0%   550   Bacteria;Proteobacteria;Gammaproteobacteria;Alteromonadales;Shewanellaceae;Shewanella;Shewanella baltica   shew-os223.sig:38729c63 NC_011663.1 Shewanella baltica OS223, complete genome
</code></pre>
<p>which indicates that about 50% of the content is <em>not</em> strain specific, and is shared with the other Shewanella in the collection.</p></div>
                <footer>
    <div class="footer-buttons">
        <div class="previous"><a href="../6-creating-a-conda-environment-with-the-latest-version-of-sourmash/" title="Example: creating a conda environment with the latest version of sourmash"><span>Previous</span></a></div>
        <div class="next"><a href="../1-downloading-sketching-and-searching-a-collection-of-FASTA-files/" title="Example: downloading sketching and searching a collection of FASTA files"><span>Next</span></a></div>
    </div>
    <div class="footer-note">
        <p>
            Built with <a href="http://www.mkdocs.org">MkDocs</a> using
            <a href="https://github.com/daizutabi/mkdocs-ivory">Ivory theme</a>.
        </p>
    </div>
</footer>
            </div>
        </main>
    </div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js"></script>
    <script src="../search/main.js"></script>
</body>

</html>